pkg/taskrunmetrics/metrics_test.go

```diff
@@ -59,7 +59,7 @@ func getConfigContext() context.Context {
 func TestUninitializedMetrics(t *testing.T) {
 	metrics := Recorder{}
 
-	if err := metrics.DurationAndCount(&v1beta1.TaskRun{}); err == nil {
+	if err := metrics.DurationAndCount(&v1beta1.TaskRun{}, nil); err == nil {
```

Comment:
thank you @khrm for adding this, is it possible to add a separate test to validate metrics count when `before` condition is not nil?

Reply:
I added tests with `before` condition.

Reply:
I added `before` condition tests.